src/audio/drc/drc.c

```diff
@@ -261,6 +302,10 @@ static int drc_process(struct processing_module *mod,
 		}
 	}
 
+	/* Control pass-though in processing function with switch control */
+	if (drc_has_config(cd))
```

Comment:
Not sure if drc_has_config() is necessary. In my point of view, the code can be
```
static bool drc_get_config_enabled(struct drc_comp_data *cd)
{
	return (cd->config) && (cd->config->params.enabled);
}
...
	/* Control pass-through in processing function with switch control */
	cd->enabled = drc_get_config_enabled(cd) && cd->enable_switch;

```
The present DRC is designed with the assumption that cd->config is always valid while in processing. The enable flag in parameter is static in our use cases, while considering the switch control support the in-param flag can be treated as the default state, and the switch flag is the dynamic signal to unmask the state if needed.

If cd->config is invalid, it makes more senses that DRC should be in disabled (pass-through) state. This is not handled by this commit.